sets.c

```diff
@@ -0,0 +1,216 @@
+/*
+    Create sets.
+
+    Copyright (C) 2016-2017 Bijan Kazemi <bkazemi@users.sf.net>
+
+    This file is part of libscanmem.
+
+    This library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/types.h>
+#include <ctype.h>
+#include <assert.h>
+#include <errno.h>
+
+#include "sets.h"
+
+#define DEFAULT_UINTLS_SZ (1000)
+
+/*
+ * Used with `qsort()`.
+ *
+ * This function uses the less elegant branching solution
+ * to deal with possible overflow.
+ */
+static int _size_t_cmp(const void *_i1, const void *_i2)
+{
+    const size_t i1 = *(const size_t *)_i1, i2 = *(const size_t *)_i2;
+
+    if      (i1 <  i2) return -1;
+    else if (i1 == i2) return  0;
+    else               return  1;
+}
+
+bool parse_uintset(const char *lptr, struct set *set, size_t maxsz)
+{
+    const char        *tok, *tmpnum = NULL, *tmpnum_end;
+    char              *tmpnum_endptr = NULL;
+    bool               got_num;
+    size_t             last_num, arr_szfilled, arr_maxsz;
```

Comment:
gcc 7 spits this:
```
sets.c: In function ‘parse_uintset’:
sets.c:54:24: warning: ‘last_num’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     size_t             last_num, arr_szfilled, arr_maxsz;
                        ^~~~~~~~
```
I know it's not actually used uninitialized, but please make it happy.

Reply:
Right, I am paranoid I am missing some execution path but I think it's OK.

Reply:
Fixed.